Marlin/src/HAL/STM32F1/build_flags.py

```diff
@@ -1,8 +1,30 @@
 from __future__ import print_function
 import sys
+import re
+
+# Hack to fetch the RX and TX buffer size from the configuration file
+def findDefinition(name):
+	f = open("Marlin/Configuration_adv.h").read().split("\n")
```

Comment:
You don't need this... ALL defines are already set on env['MARLIN_FEATURES']....

Reply:
You don't have env in this script. It's called by "!python script" in platformio's stm32f1 and it's not an extrascript

Reply:
So just change it run on PIO......

Reply:
I meant, I've tried to add an `Import("env")` on top, but it failed

Reply:
You need to change it to a extra_script, as others scripts that run inside PIO context. It should be a pre, but run after common.extra_script.

Reply:
The script makes use of an `env` variable in the bottom. So either I rewrite the bottom of the script to rename the variable to something else than `env`, but it'll be a large patch, or simple let the small function do its work.

Reply:
hey, look at the code... that script already support run on `!python ` or using extra_scripts... it's all there... don't need to rewrite anything

Reply:
To be honest, I don't understand the 2 parts in the script (`if __name__ == "__main__"` or not). So I can add another script for the 2 defines instead, but it's adding complexity, don't you think?

Reply:
`if __name__ == "__main__"` is true when this script is called alone, by command line or `!python`... it's false if it's running inside PIO as a extra script.

About the complexity: that is the reason I said to just add the definition in the build_flags for STM32F1.

Reply:
Ok, done the right way, this time I hope.